crates/oxc_minifier/src/peephole/minimize_conditions.rs

```diff
@@ -698,64 +716,76 @@ impl<'a> PeepholeOptimizations {
     ///
     /// For `target_expr` = `a`, `expr` = `a.b`, this function changes `expr` to `a?.b` and returns true.
     fn inject_optional_chaining_if_matched(
-        target_expr: &Expression<'a>,
+        target_id_name: &str,
+        expr_to_inject: &mut Expression<'a>,
         expr: &mut Expression<'a>,
         ctx: Ctx<'a, '_>,
     ) -> bool {
-        fn inject(target_expr: &Expression<'_>, expr: &mut Expression<'_>) -> bool {
+        fn inject<'aa>(
```

Comment:
`'aa` 🤨 I try to avoid inlining functions inside functions.

Reply:
The reason why this function is placed inside this function is because this function doesn't make sense without the outer function. I think we can avoid this lifetime parameter by three ways.
1. put this inner function directly in `impl<'a> PeepholeOptimizations` block as `inject_optional_chaining_if_matched_inner`
1. convert this inner function to a closure
1. convert the recursive call into a loop

Reply:
Which one should I do? or are there any better ways to do?

Reply:
I prefer `inject_optional_chaining_if_matched_inner`